src/components/chart/ChartContainer.tsx

```diff
@@ -0,0 +1,47 @@
+import * as React from 'react';
+import * as _ from 'underscore';
+
+export interface ChartContainerProps {
+    id: string;
+    renderChart: (width: number, height: number) => React.ReactNode;
+}
+
+export interface ChartBaseProps {
+    width: number;
+    height: number;
+}
+
+export const ChartContainer: React.FunctionComponent<ChartContainerProps> = (props) => {
+    const container = React.useRef<HTMLDivElement>(null);
+    useDebouncedWindowResize();
+
+    const toggleSvgs = (show: boolean) => {
+        if (container.current) {
+            const svgs = container.current.querySelectorAll('svg');
+            _.each(svgs, (svg: SVGElement) => svg.style.display = show ? 'block' : 'none');
+        }
+    };
+
+    toggleSvgs(false);
+    const {width, height} = container.current && container.current.getBoundingClientRect() || {width: 0, height: 0};
+    toggleSvgs(true);
+
+    return (
+        <div className='full-content' {..._.omit(props, 'renderChart')} ref={container}>
+            {width + height > 0 ? props.renderChart(width, height) : null}
+        </div>
+    );
+};
+
+const useForceUpdate = () => React.useState(undefined)[1];

```

Comment:
its safe to do that, it looks weird without validation

Reply:
useState always return an array `[currentState, setNewState]`. Here I just return setNewState because I don't care about the currentState